kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/GenericKubernetesResource.java

```diff
@@ -69,4 +74,55 @@ public void setAdditionalProperty(String name, Object value) {
     this.additionalProperties.put(name, value);
   }
 
+  @JsonIgnore
+  public JsonNode getAdditionalPropertiesNode() {
+    return MAPPER.convertValue(getAdditionalProperties(), JsonNode.class);
+  }
+
+  /**
+   * Allows the retrieval of field values from this Resource for the provided path segments.
+   *
+   * <p> If the path segment is of type {@link Integer}, then we assume that it is an array index to retrieve
+   * the value of an entry in the array.
+   *
+   * <p> In any other case, the path segment is assumed to be the field name to retrieve the value of from the
+   * currently traversed path object.
+   *
+   * <p> Considering the following JSON object:
+   *
+   * <pre>{@code
+   * {
+   *   "field": {
+   *     "value": 42
+   *     "list": [
+   *       {entry: 1}, {entry: 2}, {entry: 3}
+   *     ]
+   *   }
+   * }
+   * }</pre>
+   *
+   * <p> The following invocations will produce the documented results:
+   * <ul>
+   *   <li>{@code get("field", "value")} will result in {@code 42}</li>
+   *   <li>{@code get("field", "list", 1, "entry)} will result in {@code 2}</li>
+   * </ul>
+   *
+   * @param path of the field to retrieve.
+   * @param <T> type of the returned object.
+   * @return the value of the traversed path or null if the field does not exist.
+   */
+  @SuppressWarnings("unchecked")
+  public <T> T get(Object... path) {
+    Object current = getAdditionalProperties();
+    for (Object segment : path) {
+      if (segment instanceof Integer && current instanceof Collection && ((Collection<?>)current).size() > (int)segment) {
```

Comment:
I gave a thought to this while implementing.

I finally opted for the lenient approach since it had less branching in the logic implementation and I assumed it wouldn't be a problem.

However, the fact that you raised this might indicate the opposite.

Reply:
Yes, we just need to say in the javadoc that only strings and integers are expected.

Reply:
:sweat_smile: I think everything should be covered now.